src/core/Table/hooks/useSubRowFiltering.tsx

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { Hooks, IdType, Row, TableInstance } from 'react-table';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import * as defaultFilterTypes from '../../../../node_modules/react-table/src/filterTypes.js';
+
+export const useSubRowFiltering = <T extends Record<string, unknown>>(
+  showSubRows: boolean,
+) => (hooks: Hooks<T>) => {
+  hooks.useInstance.push(useInstance.bind({}, showSubRows));
+};
+
+const useInstance = <T extends Record<string, unknown>>(
+  showSubRows: boolean,
+  instance: TableInstance<T>,
+) => {
+  // There is a bug in react-table that filtered out sub-rows won't be shown after clearing the filter.
+  // Therefore saving original sub-rows separately.
```

Comment:
Would be nice to create an issue in react-table and link here.

Reply:
Added this issue to comment: https://github.com/tannerlinsley/react-table/issues/2928